README.md

```diff
@@ -1,57 +1,297 @@
+    export AZURE_SUBSCRIPTION_ID={your subscription id}
+    ```
 
-## Demo
+    > [AZURE.NOTE] On Windows, use `set` instead of `export`.
 
-A demo app is included to show how to use the project.
+7. Run the sample.
 
-To run the demo, follow these steps:
+    ```
+    bundle exec ruby example.rb
+    ```
 
-(Add steps to start up the demo)
+<a id="example"></a>
+## What is example.rb doing?
 
-1.
-2.
-3.
+This sample starts by setting up ResourceManagementClient, the resource provider clients, a resource group, and a storage account using your subscription and credentials.
 
-## Resources
+```ruby
+subscription_id = ENV['AZURE_SUBSCRIPTION_ID'] || '11111111-1111-1111-1111-111111111111' # your Azure Subscription Id
+  provider = MsRestAzure::ApplicationTokenProvider.new(
+      ENV['AZURE_TENANT_ID'],
+      ENV['AZURE_CLIENT_ID'],
+      ENV['AZURE_CLIENT_SECRET'])
+  credentials = MsRest::TokenCredentials.new(provider)
+  resource_client = Azure::ARM::Resources::ResourceManagementClient.new(credentials)
+  resource_client.subscription_id = subscription_id
+  network_client = Azure::ARM::Network::NetworkManagementClient.new(credentials)
+  network_client.subscription_id = subscription_id
+  storage_client = Azure::ARM::Storage::StorageManagementClient.new(credentials)
+  storage_client.subscription_id = subscription_id
+  compute_client = Azure::ARM::Compute::ComputeManagementClient.new(credentials)
+  compute_client.subscription_id = subscription_id
 
-(Any additional resources or related projects)
+  #
+  # Managing resource groups
+  #
+  resource_group_params = Azure::ARM::Resources::Models::ResourceGroup.new.tap do |rg|
+    rg.location = WEST_US
+  end
 
-- Link to supporting information
-- Link to similar sample
-- ...
+  # Create Resource group
+  puts 'Create Resource Group'
+  print_group resource_client.resource_groups.create_or_update(GROUP_NAME, resource_group_params)
+
+  postfix = rand(1000)
+  storage_account_name = "rubystor#{postfix}"
+  puts "Creating a premium storage account with encryption off named #{storage_account_name} in resource group #{GROUP_NAME}"
+  storage_create_params = StorageModels::StorageAccountCreateParameters.new.tap do |account|
+    account.location = WEST_US
+    account.sku = StorageModels::Sku.new.tap do |sku|
+      sku.name = StorageModels::SkuName::PremiumLRS
+      sku.tier = StorageModels::SkuTier::Premium
+    end
+    account.kind = StorageModels::Kind::Storage
+    account.encryption = StorageModels::Encryption.new.tap do |encrypt|
+      encrypt.services = StorageModels::EncryptionServices.new.tap do |services|
+        services.blob = StorageModels::EncryptionService.new.tap do |service|
+          service.enabled = false
+        end
+      end
+    end
+  end
+  print_item storage_account = storage_client.storage_accounts.create(GROUP_NAME, storage_account_name, storage_create_params)
+```
+
+<a id="vnet"></a>
+### Create a virtual network
+
```

Comment:
You may want to add a comment or two to each section, since the beginning starts with a sentence, but then it doesn't look like we "comment" much on what's expected. 
Maybe here could be something, "We create a VM so we can install MSI, this looks like a regular Azure VM creation."

Reply:
should we also add a reference to a regular VM creation example?